subsys/shell/modules/kernel_service.c

```diff
@@ -218,6 +222,66 @@ static int cmd_kernel_stacks(const struct shell *shell,
 }
 #endif
 
+static int cmd_kernel_sleep(const struct shell *sh,
+			    size_t argc, char **argv)
+{
+	ARG_UNUSED(sh);
+	ARG_UNUSED(argc);
+
+	k_msleep(strtoul(argv[1], NULL, 10));
+
+	return 0;
+}
+
+#if defined(CONFIG_LOG_RUNTIME_FILTERING)
+static int16_t log_source_id_get(const char *name)
+{
+	uint32_t count = log_src_cnt_get(CONFIG_LOG_DOMAIN_ID);
+	if (count > INT16_MAX) {
+		return -1;
+	}
+
+	for (uint32_t i = 0; i < count; i++) {
+		if (strcmp(log_source_name_get(CONFIG_LOG_DOMAIN_ID, i), name)
+		    == 0) {
+			return (int16_t)i;
+		}
+	}
+	return -1;
+}
+
+static int cmd_kernel_log_level_set(const struct shell *sh,
+				    size_t argc, char **argv)
+{
+	ARG_UNUSED(argc);
+	ARG_UNUSED(argv);
+	int err = 0;
+
+	uint8_t severity = shell_strtoul(argv[2], 10, &err);
+
+	if (err) {
+		shell_error(sh, "Unable to parse log severity");
+
+		return -ENOEXEC;
+	}
+
+	if (severity > 5) {
```

Comment:
Do we have a properly defined value somewhere instead of the magical number `5`?

Reply:
yep, 5 was wrong anyways, it was 4 oops